core/src/main/java/org/mule/runtime/core/privileged/event/PrivilegedEvent.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.privileged.event;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.connector.ReplyToHandler;
+import org.mule.runtime.core.api.event.BaseEvent;
+import org.mule.runtime.core.api.event.BaseEventContext;
+import org.mule.runtime.core.api.event.MuleSession;
+import org.mule.runtime.core.api.source.MessageSource;
+import org.mule.runtime.core.api.transformer.MessageTransformerException;
+
+import java.nio.charset.Charset;
+
+/**
+ * Allows access to the privileged behavior of the {@link BaseEvent}.
```

Comment:
There is no privileged behaviour in BaseEvent, probably you want to refer to the event implementation instead